griptape/tools/aws_s3_client/tool.py

```diff
@@ -174,6 +174,40 @@ def upload_content_to_s3(self, params: dict) -> ErrorArtifact | InfoArtifact:
         except Exception as e:
             return ErrorArtifact(f"error uploading objects to the bucket: {e}")
 
+    @activity(config={
+        "description": "Can be used to download an object from an AWS S3 bucket",
+        "schema": Schema(
+            {
+                Literal(
+                    "bucket_name",
+                    description="The name of the S3 bucket to download from."
+                ): str,
+                Literal(
+                    "object_key",
+                    description="The object key name to download."
+                ): str
+            }
+        )
+    })
+    def download_object(self, params: dict) -> ErrorArtifact | TextArtifact | BlobArtifact:
+        bucket_name = params["values"]["bucket_name"]
+        object_key = params["values"]["object_key"]
+
+        try:
+            obj = self.s3_client.get_object(
+                Bucket=bucket_name,
+                Key=object_key
+            )
+
+            # Return a TextArtifact if the object appears to be text.
+            if "text" in obj["ContentType"]:
+                return TextArtifact(obj["Body"].read())
+
+            return BlobArtifact(obj["Body"].read())
```

Comment:
Here, we attempt to interpret the object as text if the ContentType implies it's text. Otherwise, we handle it as a BlobArtifact.